File: 2444-Count_Subarrays_With_Fixed_Bounds/Count_Subarrays_With_Fixed_Bounds.go
package countsubarrayswithfixedbounds

func countSubarrays(nums []int, minK int, maxK int) int64 {
	var count int64
	minP := -1
	maxP := -1
	leftBound := -1

	for i, val := range nums {
		if val < minK || val > maxK {
			leftBound = i
		}

		if val == minK {
			minP = i
		}

		if val == maxK {
			maxP = i
		}

		distance := min(minP, maxP) - leftBound
		if distance > 0 {
			count += int64(distance)
		}
	}

	return count
}

func min(a int, b int) int {
	if a < b {
		return a
	}

	return b
}

File: 2444-CountSubarraysWithFixedBounds/CountSubarraysWithFixedBounds.java

class CountSubarraysWithFixedBounds {
    class Solution {
        public long countSubarrays(int[] nums, int minK, int maxK) {
            long count = 0l;
            int minP = -1, maxP = -1, leftBound = -1;
            
            for(int i = 0 ; i < nums.length; i++){
                if(nums[i] < minK || nums[i] > maxK)
                    leftBound = i;
                
                if(nums[i] == minK) minP = i;
                if(nums[i] == maxK) maxP = i;
                
                count += Math.max(0, Math.min(minP, maxP) - leftBound);
            }
            
            
            return count;
        }
    }
}
